File: BurgerBuilderBackend/BurgerBuilder.DTOs/Classes/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace BurgerBuilder.DTOs.Classes
{
    public class Order
    {
        public int ID { get; set; }
        [Required]
        public Ingridients Ingridients { get; set; }
        public decimal Price { get; set; }
        [Required]
        public Customer Customer { get; set; }
        [Required]
        public string DeliveryMethod { get; set; }
    }
}

File: burger-builder/src/Types/Order.ts
import { BurgerInnerIngridientsDictionary } from "./BurgerInnerIngridientsDictionary";
import ICustomer from "./Customer";
import { OrderDeliveryMethod } from "./OrderDeliveryMethod";

export default interface IOrder {
    id: number,
    deliveryMethod: OrderDeliveryMethod,
    price: number,
    ingridients: BurgerInnerIngridientsDictionary,
    customer: ICustomer
 }